app/script/view_model/content/InputBarViewModel.js

```diff
@@ -95,17 +95,22 @@ z.viewModel.content.InputBarViewModel = class InputBarViewModel {
 
     this.richTextInput = ko.pureComputed(() => {
       const input = this.input();
-      const mentions = this.parseForMentions(input, this.conversationEntity().participating_user_ets());
-      const pieces = mentions.reverse().reduce(
-        (currentPieces, mention) => {
-          const currentPiece = currentPieces.shift();
-          currentPieces.unshift(currentPiece.substr(mention.end));
-          currentPieces.unshift(currentPiece.substr(mention.start, mention.end - mention.start));
-          currentPieces.unshift(currentPiece.substr(0, mention.start));
-          return currentPieces;
-        },
-        [input]
-      );
+      const participatingUserEntities = this.conversationEntity().participating_user_ets();
+
+      const pieces = this.parseForMentions(input, participatingUserEntities)
+        .reverse()
+        .reduce(
+          (currentPieces, mentionEntity) => {
+            const {end: mentionEnd, start: mentionStart} = mentionEntity;
+
+            const currentPiece = currentPieces.shift();
+            currentPieces.unshift(currentPiece.substr(mentionEnd));
+            currentPieces.unshift(currentPiece.substr(mentionStart, mentionEnd - mentionStart));
```

Comment:
Now that we have an entity for the messages, we could put ```static get length() {return this.end-this.start}``` in there. WDYT?

Reply:
Sure. Good idea. Will add that.